run.ts

```diff
@@ -63,37 +65,31 @@ const launchDocker: SearchEngineLauncherFunction = async ({
   port,
   options,
 }) => {
-  const Image =
-    engine === 'elasticsearch'
-      ? 'elastic/elasticsearch:8.6.2'
-      : 'opensearchproject/opensearch:2.11.0'
-  console.log('Launching Docker container', Image)
-  const docker = new Dockerode()
-  const container = await docker.createContainer({
-    Env: [...options, 'path.data=/var/lib/search', 'path.logs=/var/log/search'],
-    HostConfig: {
-      AutoRemove: true,
-      Mounts: [
-        { Source: dataDir, Target: '/var/lib/search', Type: 'bind' },
-        { Source: logsDir, Target: '/var/log/search', Type: 'bind' },
-      ],
-      PortBindings: {
-        [`${port}/tcp`]: [{ HostIP: '127.0.0.1', HostPort: `${port}` }],
-      },
-    },
-    Image,
-  })
-  const stream = await container.attach({ stream: true, stderr: true })
-  stream.pipe(process.stderr)
-  await container.start()
+  const argv = {
+    dataDir,
+    logsDir,
+    engine,
+    port,
+    options,
+  }
+  const __filename = fileURLToPath(import.meta.url)
+  const __dirname = path.dirname(__filename)
+  const subprocess = fork(`${__dirname}/launchSearch.js`, [
```

Comment:
```suggestion
  join(import.meta.dirname, 'launchSearch.js`, [
```

And of course add `import { join } from 'node:path'`.

Reply:
<img width="880" alt="Screenshot 2024-06-13 at 3 27 50 PM" src="https://github.com/nasa-gcn/architect-plugin-search/assets/6545874/4cddd770-5faa-4455-bdb8-748cb77aff30">

Reply:
ah, I was one version behind. so if your local is not using the proper node version it's undefined.

Reply:
do you know if most people are using nvm? because we can prevent it from being undefined because of the wrong version if we add an .nvmrc file. That tells nvm what version to use for the project. But that would only be effective if people are using nvm.
Because if I use import.meta.dirname and they are not at the proper version, doing it this way will break whereas the other way would work regardless of node version.

Reply:
No, I don't think we can expect that everyone is using NVM. What are you trying to do? Is `importlib.meta.dirname` only supported in recent Node.js versions? What is the minimum version that you need?

Reply:
upon further testing, the way that requires a node version of 20.11 doesn't seem to be an option right now. 
I just ran into something interesting. As soon as I updated my node version to be 20.11.0, in order to terminate the process I started having to hit control + c twice. I changed my architect-plugin-search branch back to main, re-ran `npm prepare` and ensured that the package had updated in gcn. I then ran GCN at main after an `npm install`. I had to control + c twice to exit the process. 

<img width="554" alt="Screenshot 2024-06-13 at 4 32 38 PM" src="https://github.com/nasa-gcn/architect-plugin-search/assets/6545874/7187ba6a-1a3e-4b9e-a634-be75b89eeaa8">

when I roll my version back to v20.9.0 I can quit out of the process with one control + c. 

<img width="547" alt="Screenshot 2024-06-13 at 4 37 30 PM" src="https://github.com/nasa-gcn/architect-plugin-search/assets/6545874/c0c1a9d7-94bf-4e15-b47e-1ce5f3c93c25">

I haven't looked into what is making the process stick around in version v20.11.0 yet, so I don't know why it's doing it. Just that it happens with both branches on main if my version is 20.11.0.

Reply:
> upon further testing, the way that requires a node version of 20.11 doesn't seem to be an option right now.

Alright. `import.meta.url` may be a better option, then. That has been supported at least as far back as Node.js 18. Please use `import.meta.url`.

Reply:
that brings us back to this part of my other comment (cutting and pasting it here since it's relevant in this thread too):

 `import.meta.url` is `architect-plugin-search/index.js` so that can't be used directly. Here is what I'm doing there and why it needs to have multiple steps:

1. import.meta.url: file:///Users/courey/dev/architect-plugin-search/index.js this is getting the name of the file that is currently running. in this case it's index.js. I don't want index.js, I want launchSearch.js. 
2. so we then pass that into fileURLToPath so __filename would then be: /Users/courey/dev/architect-plugin-search/index.js. This is preparing it so that we can use path.
3. I don't want the file, just the directory. Since we converted that to a path with fileURLToPath we can use path.dirname() to get just the path of the directory instead of the file that is running. that makes __dirname: /Users/courey/dev/architect-plugin-search
4. I add /launchSearch.js to the end of that path to have the proper path and proper file name.

Reply:
We can make this a lot simpler. There is no need to compute the path. Just bundle a single entry point with esbuild, as we did originally. Then launch the subprocess like this:

```ts
const subprocess = fork(import.meta.url, 'launch-docker-subprocess', JSON.stringify({
    dataDir,
    logsDir,
    engine,
    port,
    options,
  })
```

At file scope, add this code:

```ts
const [, , command, jsonifiedArgs] = process.argv
if (command === 'launch-docker-subprocess') {
  // put everything that you want the subprocess to do in here:
  // launch the container, attach signal handlers, wait for container to exit
}
```